syntax = "proto3";

package imageservice;

service ImageService {
  rpc UploadImage (ImageUploadRequest) returns (ImageUploadResponse);
  rpc UploadImageStream (stream ImageChunk) returns (ImageUploadResponse);
}

message ImageUploadRequest {
  string image_name = 1;
  bytes image_data = 2;
}

message ImageChunk {
  string image_name = 1;
  bytes chunk_data = 2;
}

message ImageUploadResponse {
  string message = 1;
  string file_path = 2;
}
